frontend/packages/shared/src/components/StudioNumberInput/StudioNumberInput.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+import { Textfield } from '@digdir/design-system-react';
+import { useTranslation } from 'react-i18next';
+
+export interface StudioNumberInputProps {
+  description: string;
+  onChange: (value: number) => void;
+}
+
+export const StudioNumberInput = ({ description, onChange }: StudioNumberInputProps) => {
+  const { t } = useTranslation();
+  const [inputValue, setInputValue] = useState('');
+  const [showError, setShowError] = useState(false);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const input = e.target.value;
+    setInputValue(input);
+    onChange(Number(input.replace(',', '.')));
+  };
+
+  const validateNumber = (value: string) => {
+    const numberRegex = /^[0-9]+([.,][0-9]*)?$/;
+    return showError && !numberRegex.test(value) ? t('validation_errors.numbers_only') : undefined;
+  };
+
+  const handleBlur = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const input = e.target.value;
+    setShowError(true);
+    if (input === '') {
+      setShowError(false);
+    }
+  };
+
+  return (
+    <div>
+      <Textfield
+        description={description}
+        value={inputValue}
+        onChange={handleInputChange}
+        error={validateNumber(inputValue)}
+        onFocus={() => setShowError(false)}
```

Comment:
Jeg tror vi kan droppe denne. Når feilmeldingen først vises, er det ingen grunn til å skjule den igjen før brukeren har rettet feilen.

Reply:
jeg har lagt `  onFocus={() => setHasBeenBlurred(false)} ` som fikser en issue jeg fant  og det er når brukeren skriver bokstaver to ganger så kommer feil melding med første bokstav bruker skriver:
Forklaring:
- skrive bokstaver og trykk ut så kommer feil melding, den er riktig og som skal.
- Fjern bokstavene som du skriver og trykke ut, så kom igjen og skrive bokstaver , her kommer feil melding med første bokstav man skriver.

Reply:
Jeg tenker det er greit at det oppfører seg slik. Da har man allerede prøv å fylle ut feltet en gang. Problemet med å deaktivere dette på `onFocus` er at feilmeldingen forsvinner med en gang brukeren fokuserer på feltet, så det kan oppfattes som at det ikke er feil likevel.

En alternativ fremgangsmåte er å legge til en statevariabel som heter `hasStartedTyping` eller noe slikt og styre den ved hjelp av `onChange`, og i tillegg sette `hasBeenBlurred` til `false` når inputfeltet er tomt;
```
  const handleInputChange = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => {
      const input = e.target.value;
      setInputValue(input);
      setHasStartedTyping(!!input.length);
      if (!input.length) setHasBeenBlurred(false);
      onChange(convertStringToNumber(input));
    },
    [setInputValue, onChange],
  );
```
Så må vi også sjekke at `hasStartedTyping` er `true` for å vise feilmeldingen.
Dette løser problemet du nevnte, i tillegg til at feilmeldingen blir stående når man fokuserer igjen. Feilmeldingsvisningen blir heller ikke aktivert bare av at brukeren klikker seg inn i feltet og ut igjen. På den negative siden begynner koden å se ganske rotete ut hvis vi innfører dette på den måten jeg beskrev her. Kanskje det er bedre å abstrahere ut dette ved hjelp av [`useReducer`](https://react.dev/reference/react/useReducer)?